cryptol.cabal

```diff
@@ -58,6 +58,7 @@ library
                        GraphSCC          >= 1.0.4,
                        heredoc           >= 0.2,
                        libBF             >= 0.6 && < 0.7,
+                       libffi,
```

Comment:
It would be worth guarding this dependency behind a Cabal flag. The reason is that Windows doesn't come with `libffi` by default, so most Windows users who try to naïvely install this would run into a build failure. Guarding it behind a Cabal flag would ensure that the solver can detect this situation and avoid depending on `libffi` if the user hasn't installed it.

The flip side to doing this is that we'd need a dummy implementation of the FFI that always fails if the user doesn't have `libffi` installed.